src/outputs/restart.cpp

```diff
@@ -17,30 +17,582 @@
+        tmpData[i] = xmin[2];
+        i++;
+      }
+      pmb = pmb->next;
+    }
+    local_count[1] = global_count[1] = pm->ndim;
+    WRITEH5SLABDOUBLE("xmin", tmpData, gBlocks, local_start, local_count, global_count,
+                      property_list);
+    delete[] tmpData;
+  }
+
+  // write Block ID
+  {
+    // LOC.lx1,2,3
+    hsize_t n;
+    int i;
+
+    n = 3;
+    auto *tmpLoc = new int64_t[num_blocks_local * n];
+    local_count[1] = global_count[1] = n;
+    local_count[0] = num_blocks_local;
+    global_count[0] = max_blocks_global;
+    pmb = pm->pblock;
+    i = 0;
+    while (pmb != nullptr) {
+      tmpLoc[i++] = pmb->loc.lx1;
+      tmpLoc[i++] = pmb->loc.lx2;
+      tmpLoc[i++] = pmb->loc.lx3;
+      pmb = pmb->next;
+    }
+    WRITEH5SLABI64("loc.lx123", tmpLoc, gBlocks, local_start, local_count, global_count,
+                   property_list);
+    delete[] tmpLoc;
+
+    // (LOC.)level, GID, LID, cnghost, gflag
+    n = 5;
+    auto *tmpID = new int[num_blocks_local * n];
+    local_count[1] = global_count[1] = n;
+    local_count[0] = num_blocks_local;
+    global_count[0] = max_blocks_global;
+    pmb = pm->pblock;
+    i = 0;
+    while (pmb != nullptr) {
+      tmpID[i++] = pmb->loc.level;
+      tmpID[i++] = pmb->gid;
+      tmpID[i++] = pmb->lid;
+      tmpID[i++] = pmb->cnghost;
+      tmpID[i++] = pmb->gflag;
+      pmb = pmb->next;
+    }
+    WRITEH5SLABI32("loc.level-gid-lid-cnghost-gflag", tmpID, gBlocks, local_start,
+                   local_count, global_count, property_list);
+    delete[] tmpID;
+  }
+
+  // close locations tab
+  H5Gclose(gBlocks);
+
+  // write variables
+
+  // write variables
+  // create persistent spaces
+  local_count[1] = nx3;
+  local_count[2] = nx2;
+  local_count[3] = nx1;
+  local_count[4] = 1;
+
+  global_count[1] = nx3;
+  global_count[2] = nx2;
+  global_count[3] = nx1;
+  global_count[4] = 1;
+
+  hid_t local_DSpace = H5Screate_simple(5, local_count, NULL);
+  hid_t global_DSpace = H5Screate_simple(5, global_count, NULL);
+
+  // while we could do this as n variables and load all variables for
+  // a block at one time, this is memory-expensive.  I think it is
+  // well worth the multiple iterations through the blocks to load up
+  // one variable at a time.  Besides most of the time will be spent
+  // writing the HDF5 file to disk anyway...
+  // If I'm wrong about this, we can always rewrite this later.
+  // Sriram
+
+  const hsize_t varSize = nx3 * nx2 * nx1;
+
+  auto ciX = ContainerIterator<Real>(
+      pm->pblock->real_containers.Get(),
+      {parthenon::Metadata::Independent, parthenon::Metadata::Restart}, true);
+  for (auto &vwrite : ciX.vars) { // for each variable we write
+    const std::string vWriteName = vwrite->label();
+    hid_t vLocalSpace, vGlobalSpace;
+    pmb = pm->pblock;
+    const hsize_t vlen = vwrite->GetDim(4);
+    local_count[4] = global_count[4] = vlen;
+    Real *tmpData = new Real[varSize * vlen * num_blocks_local];
```

Comment:
`std::vector`...

Reply:
Didn't want to use std::vector because this is passed to hdf interface as (void *).  Unnecessary casts back and forth.

Reply:
Just pass it in to hdf5 as `tmpData.data()` if it's a vector. No cast necessary. Or if you're me and evil, `&(tmpData[0])`

I strongly prefer using a vector because I do not like `new` and `delete` floating around. It's asking for a memory leak.